cav_msgs/msg/TrafficControlDetail.msg

```diff
@@ -0,0 +1,174 @@
+#	direction ENUMERATED {forward, reverse},
+#	lataffinity ENUMERATED {left, right},
+#	latperm SEQUENCE (SIZE(2)) OF ENUMERATED {none, permitted, passing-only, emergency-only},
+#	parking ENUMERATED {no, parallel, angled},
+#	minspeed INTEGER (0..1023), -- tenths of m/s
+#	maxspeed INTEGER (0..1023), -- tenths of m/s
+#	minhdwy INTEGER (0..2047), -- tenths of meters
+#	maxvehmass INTEGER (0..65535), -- kg
+#	maxvehheight INTEGER (0..127), -- tenths of meters
+#	maxvehwidth INTEGER (0..127), -- tenths of meters
+#	maxvehlength INTEGER (0..1023), -- tenths of meters
+#	maxvehaxles INTEGER (2..15),
+#	minvehocc INTEGER (1..15), 
+#	...
+#}
+
+
+uint8 choice
+
+# enumeration values for choice:
+
+uint8 SIGNAL_CHOICE=0
+uint8 STOP_CHOICE=1
+uint8 YIELD_CHOICE=2
+uint8 NOTOWING_CHOICE=3
+uint8 RESTRICTED_CHOICE=4
+uint8 CLOSED_CHOICE=5
+uint8 CHAINS_CHOICE=6
+uint8 DIRECTION_CHOICE=7
+uint8 LATAFFINITY_CHOICE=8
+uint8 LATPERM_CHOICE=9
+uint8 PARKING_CHOICE=10
+uint8 MINSPEED_CHOICE=11
+uint8 MAXSPEED_CHOICE=12
+uint8 MINHDWY_CHOICE=13
+uint8 MAXVEHMASS_CHOICE=14
+uint8 MAXVEHHEIGHT_CHOICE=15
+uint8 MAXVEHWIDTH_CHOICE=16
+uint8 MAXVEHLENGTH_CHOICE=17
+uint8 MAXVEHAXLES_CHOICE=18
+uint8 MINVEHOCC_CHOICE=19
+
+# signal ::= OCTET STRING SIZE(0..63)
+uint8 signal
+
+#closed ENUMERATED {open, closed, taperleft, taperright, openleft, openright}
+
+uint8 closed
+
+# enumeration values for closed:
+
+unit8 OPEN=0
+
+unit8 CLOSED=1
+
+unit8 TAPERLEFT=2
+
+unit8 TAPERRIGHT=3
+
+unit8 OPENLEFT=4
+
+unit8 OPENRIGHT=5
+
+#chains ENUMERATED {no, permitted, required}
+
+uint8 chains
+
+# enumeration values for chains:
+
+unit8 NO=0
+
+unit8 PERMITTED=1
+
+unit8 REQUIRED=2
+
+
+#direction ENUMERATED {forward, reverse}
+
+uint8 direction
+
+# enumeration values for direction:
+
+unit8 FORWARD=0
+
+unit8 REVERSE=1
+
+#lataffinity ENUMERATED {left, right}
+
+uint8 lataffinity
+
+# enumeration values for lataffinity:
+
+unit8 LEFT=0
+
+unit8 RIGHT=1
+
+#latperm SEQUENCE (SIZE(2)) OF ENUMERATED {none, permitted, passing-only, emergency-only}
+
+uint8[2] latperm
+
+# enumeration values for latperm:
+
+unit8 NONE=0
+
+unit8 PERMITTED=1
+
+unit8 PASSINGONLY=2
+
+unit8 EMERGENCYONLY=3
+
+#parking ENUMERATED {no, parallel, angled}
+
+uint8 parking
+
+# enumeration values for parking:
+
+unit8 NO=0
+
+unit8 PARALLEL=1
+
+unit8 ANGLED=1
+
+#minspeed ::= INTEGER (0..1023), -- tenths of m/s
+float64 minspeed
+
+#maxspeed ::= INTEGER (0..1023), -- tenths of m/s
+float64 maxspeed
+
+#minhdwy ::= INTEGER (0..2047), -- tenths of meters
+float64 minhdwy
+
+#maxvehmass ::= INTEGER (0..65535), -- kg
+float64 maxvehmass
+
+#maxvehheight ::= INTEGER (0..127), -- tenths of meters
+float64 maxvehheight
+
+#maxvehwidth ::= INTEGER (0..127), -- tenths of meters
+float64 maxvehwidth
+
+#maxvehlength ::= INTEGER (0..1023), -- tenths of meters
+float64 maxvehlength
+
+#maxvehaxles ::= INTEGER (2..15)
+float64 maxvehaxles
+
+#minvehocc ::= INTEGER (1..15)
+float64 minvehocc
+
```

Comment:
I think this also should be uint8. I am assuming it means min veh occurences. Same for j2735_msgs